src/test/java/lambda/part2/exercise/FunctionCombinationExercise.java

```diff
@@ -56,25 +56,31 @@ public void personHasNotEmptyLastNameAndFirstName1() {
     // negate: (T -> boolean) -> (T -> boolean)
     private <T> Predicate<T> negate(Predicate<T> test) {
         // TODO
-        throw new UnsupportedOperationException();
+        return p -> !test.test(p);
+        //throw new UnsupportedOperationException();
     }
 
     // TODO
     // and: (T -> boolean, T -> boolean) -> (T -> boolean)
     private <T> Predicate<T> and(Predicate<T> t1, Predicate<T> t2) {
         // TODO
-        throw new UnsupportedOperationException();
+        return p -> {
+            boolean b1 = t1.test(p);
+            boolean b2 = t2.test(p);
+            return b1 && b2;

```

Comment:
`&&` is a lazy operation (`short-circuit operators`), unlike `&`. With additional local variable you are making it eager.

See [Short-circuit evaluation](https://en.wikipedia.org/wiki/Short-circuit_evaluation)